src/main/java/moimoi/gui/DialogBox.java

```diff
@@ -0,0 +1,82 @@
+package moimoi.gui;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.geometry.Pos;
+import javafx.scene.Node;
+import javafx.scene.control.Label;
+import javafx.scene.image.Image;
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
+/**
+ * Represents a dialog box consisting of an ImageView to represent the speaker's face
+ * and a label containing text from the speaker.
+ */
+public class DialogBox extends HBox {
+
+    @FXML
+    private Label dialog;
+    @FXML
+    private ImageView displayPicture;
+
+    /**
+     * Constructs a dialog box with the specified image of and text from a speaker.
+     *
+     * @param text Text from the speaker.
+     * @param img Image of the speaker.
+     */
+    private DialogBox(String text, Image img) {
+        try {
+            FXMLLoader fxmlLoader = new FXMLLoader(MainWindow.class.getResource("/view/DialogBox.fxml"));
+            fxmlLoader.setController(this);
+            fxmlLoader.setRoot(this);
+            fxmlLoader.load();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        this.dialog.setText(text);
+        this.displayPicture.setImage(img);
+    }
+
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
+        ObservableList<Node> tmp = FXCollections.observableArrayList(this.getChildren());
+        Collections.reverse(tmp);
+        getChildren().setAll(tmp);
+        setAlignment(Pos.TOP_LEFT);
+    }
+
+    /**
+     * Returns a dialog box with the specified image of and text from the user.
+     *
+     * @param text Text from the user.
+     * @param img Image of the user.
+     * @return Dialog box with the image of and text from the user.
+     */
+    public static DialogBox getUserDialog(String text, Image img) {
+        return new DialogBox(text, img);
+    }
+
+    /**
+     * Returns a dialog box with the specified image of and text from MoiMoi.
+     *
+     * @param text Text from MoiMoi.
+     * @param img Image of MoiMoi.
+     * @return Dialog box with the image of and text from MoiMoi.
+     */
+    public static DialogBox getMoiMoiDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
```

Comment:
Likewise over here, can give a better name for yourself to know what it is. Can be var dialogBox